home/tests/test_api.py

```diff
@@ -244,118 +254,142 @@ def test_pagination(self):
+        assert len(var_content) == 1
+        assert var_content[0]["profile_field"] == "gender"
+        assert var_content[0]["value"] == "female"
+        assert var_content[0]["message"] == "Test Title - female variation"
 
-        self.assertEqual(PageView.objects.count(), 1)
+        assert PageView.objects.count() == 1
         view = PageView.objects.last()
-        self.assertEqual(view.message, 1)
+        assert view.message == 1
 
-    def test_whatsapp_body(self):
+    def test_whatsapp_body(self, uclient):
         """
         Should have the WhatsApp specific fields included in the body; if it's a
         template, what's the template name, the text body of the message.
+
+        FIXME:
+         * This should probably be in a class without the content setup.
         """
         ContentPage.objects.all().delete()
         page = create_page(
             is_whatsapp_template=True, whatsapp_template_name="test_template"
         )
 
         # it should return the correct details
-        response = self.client.get(f"/api/v2/pages/{page.id}/?whatsapp")
+        response = uclient.get(f"/api/v2/pages/{page.id}/?whatsapp")
         content = response.json()
-        self.assertTrue(content["body"]["is_whatsapp_template"])
-        self.assertEqual(content["body"]["whatsapp_template_name"], "test_template")
-        self.assertEqual(
-            content["body"]["text"]["value"]["message"], "Test WhatsApp Message 1"
-        )
+        assert content["body"]["is_whatsapp_template"]
+        assert content["body"]["whatsapp_template_name"] == "test_template"
+        assert content["body"]["text"]["value"]["message"] == "Test WhatsApp Message 1"
 
-    def test_detail_view_no_content_page(self):
+    def test_detail_view_no_content_page(self, uclient):
+        """
+        FIXME:
+         * It's unclear what the intended result is.
+         * Should the response code be a success?
+        """
         # it should return the validation error for content page that doesn't exist
-        response = self.client.get("/api/v2/pages/1/")
+        response = uclient.get("/api/v2/pages/1/")
         content = response.json()
 
-        self.assertEqual(content, {"page": ["Page matching query does not exist."]})
-        self.assertEqual(content.get("page"), ["Page matching query does not exist."])
+        assert content == {"page": ["Page matching query does not exist."]}
+        assert content.get("page") == ["Page matching query does not exist."]
 
 
-class WhatsAppMessagesTestCase(TestCase):
-    def setUp(self):
-        self.user_credentials = {"username": "test", "password": "test"}
-        self.user = get_user_model().objects.create_user(**self.user_credentials)
-        self.client.login(**self.user_credentials)
+@pytest.mark.django_db
+class TestWhatsAppMessages:
+    """
+    FIXME:
+     * Should some of the WhatsApp tests from TestPagination live here instead?
+    """
```

Comment:
Yes, I think it's probably a good idea to move tests like `test_whatsapp_body` in here

Reply:
Done.